http-gateways/REDIRECTS_FILE.md

```diff
@@ -0,0 +1,173 @@
+
+This can be used, for example, to enable URL rewriting for hosting a single-page application, to redirect invalid URLs to a pretty 404 page, or to avoid  [link rot](https://en.wikipedia.org/wiki/Link_rot) when moving to IPFS-based website hosting.
+
+# Table of Contents
+
+- [File Name and Location](#file-name-and-location)
+- [File Format](#file-format)
+  - [From](#from)
+  - [To](#to)
+  - [Status](#status)
+  - [Placeholders](#placeholders)
+  - [Splat](#splat)
+- [Evaluation](#evaluation)
+  - [Subdomain or DNSLink Gateways](#subdomain-or-dnslink-gateways)
+  - [Order](#order)
+  - [No Forced Redirects](#no-forced-redirects)
+- [Error Handling](#error-handling)
+- [Security](#security)
+- [Appendix: notes for implementors](#appendix-notes-for-implementors)
+  - [Test fixtures](#test-fixtures)
+
+# File Name and Location
+
+The Redirects File MUST be named `_redirects` and stored underneath the root CID of the web site.
+
+# File Format
+
+The Redirects File MUST be a text file containing one or more lines with the following format (brackets indication optionality).
+
+```
+from to [status]
+```
+
+Any line beginning with `#` will be treated as a comment and ignored.
+
+Lines MUST be terminated by either `\n` or `\r\n`.
+
+## From
+
+The path to redirect from.
+
+## To
+
+The URL or path to redirect to.
+
+## Status
+
+An optional integer specifying the HTTP status code to return from the request.  Supported values are:
+
+- `200` - OK
+  - Redirect will be treated as a rewrite, returning OK without changing the URL in the browser.
+- `301` - Permanent Redirect (default)
+- `302` - Found (commonly used for Temporary Redirect)
+- `303` - See Other (replacing PUT and POST with GET)
+- `307` - Temporary Redirect (explicitly preserving body and HTTP method of original request)
+- `308` - Permanent Redirect (explicitly preserving body and HTTP method of original request)
+- `404` - Not Found
+  - Useful for redirecting invalid URLs to a pretty 404 page.
+- `410` - Gone
+- `451` - Unavailable For Legal Reasons  
+
+## Placeholders
+
+Placeholders are named variables that can be used to match path segments in the `from` path and inject them into the `to` path.
+
+For example:
+
+```
+/posts/:month/:day/:year/:slug  /articles/:year/:month/:day/:slug
+```
+
+This rule will redirect a URL like `/posts/06/15/2022/hello-world` to `/articles/2022/06/15/hello-world`.
+
+### Splat
+
+If a `from` path ends with an asterisk (i.e. `*`), the remainder of the `from` path is slurped up into the special `:splat` placeholder, which can then be injected into the `to` path.
+
+For example:
+
+```
+/posts/* /articles/:splat
+```
+
+This rule will redirect a URL like `/posts/2022/06/15/hello-world` to `/articles/2022/06/15/hello-world`.
+
+Splat logic MUST only apply to a single trailing asterisk, as this is a greedy match, consuming the remainder of the path.
+
```

Comment:
## Comment

A line starting with `#` is going to be considered a comment, and ignored at evaluation time.

For example:

```
# Redirect home to index.html
/home /index.html 301
```

is functionally equivalent to

```
/home /index.html 301
```

Reply:
the feature is supported by Netlify and Cloudflare Pages implementation of `_redirects` file

Reply:
@thibmeu I do have the following in the File Format section.

> Any line beginning with `#` will be treated as a comment and ignored.

Are you suggesting I remove that and call it out more prominently like this?

Reply:
yes, that's what I am suggesting. thanks for making sure there is more context.

Reply:
👍 I have now given both comments and line termination their own heading.  Thanks!